src/shared/components/post/post-listing.tsx

```diff
@@ -1153,20 +1153,20 @@ export class PostListing extends Component<PostListingProps, PostListingState> {
             ) : (
               <>
                 <button
-                  className="d-inline-block mr-1 btn btn-link btn-animate text-muted py-0"
+                  className="post-listing__btn post-listing__btn--transfc-confirm d-inline-block mr-1 btn btn-link btn-animate text-muted py-0"
```

Comment:
`transfc` = transfer community? The class name's already so long, I see no reason to *not* just write the rest of the words.

Reply:
This is what I mean about planning. Will this specific button really be styled much differently from other buttons? Does it need a modifier?

All these names are chosen with no intention other than making sure everything has a class name on it. These are extremely likely to change, breaking whatever user themes are created in the meantime. Then you have a bunch of user themers getting upset at the project for "breaking" their themes.

This is not what CSS is for.

Reply:
@alectrocute It's extremely messy and doesn't serve the larger goals of the project, these are only serving as sledgehammer hooks for custom theming. Obviously this isn't my project, but I just think this whole thing is greatly misguided.

Reply:
I agree with you on all counts, @jsit.

Reply:
@djvs It seems like the consensus is that this is not a good idea, and theme designers are going to come complaining to us every time their themes break when we change things around.

Is it so bad to generate bootstrap-compatible themes for site customization?

Reply:
That's the product of two constraints:
  * Somebody wanting to style that particular button
  * Using BEM 

And sort of additionally:
* jsit's earlier request to have generic and then modifier substrings for stuff like buttons - `post-listing__btn post-listing__btn--transfc-confirm".  Specifically, it's:

Block: post-listing

Element: btn

Modifier: transfc-confirm

The verbosity is my problem with BEM, but without it you are indeed stuck with "confirm-transfc" or whatever.  That was basically my initial approach with slightly more namespacing.